search-service/src/test/kotlin/com/egm/stellio/search/service/TemporalEntityServiceTests.kt

```diff
@@ -0,0 +1,195 @@
+            attributeName = "https://uri.etsi.org/ngsi-ld/notification",
+            attributeValueType = TemporalEntityAttribute.AttributeValueType.ANY
+        )
+        val attributeAndResultsMap = mapOf(
+            temporalEntityAttribute to listOf(
+                AttributeInstanceResult(
+                    attributeName = "https://uri.etsi.org/ngsi-ld/notification",
+                    value = "urn:ngsi-ld:Beehive:1234",
+                    observedAt = ZonedDateTime.parse("2020-03-25T08:29:17.965206Z"),
+                    payload = buildAttributeInstancePayload(
+                        "urn:ngsi-ld:Beehive:1234",
+                        ZonedDateTime.parse("2020-03-25T08:29:17.965206Z"),
+                        null,
+                        "urn:ngsi-ld:Beehive:notification:1234".toUri()
+                    )
+                ),
+                AttributeInstanceResult(
+                    attributeName = "https://uri.etsi.org/ngsi-ld/notification",
+                    value = "urn:ngsi-ld:Beehive:5678",
+                    observedAt = ZonedDateTime.parse("2020-03-25T08:33:17.965206Z"),
+                    payload = buildAttributeInstancePayload(
+                        "urn:ngsi-ld:Beehive:5678",
+                        ZonedDateTime.parse("2020-03-25T08:33:17.965206Z"),
+                        null,
+                        "urn:ngsi-ld:Beehive:notification:4567".toUri()
+                    )
+                )
+            )
+        )
+
+        val temporalEntity = temporalEntityService.buildTemporalEntity(
+            "urn:ngsi-ld:Subscription:1234".toUri(),
+            attributeAndResultsMap,
+            listOf(NGSILD_CORE_CONTEXT),
+            false
+        )
+        assertTrue(
+            serializeObject(temporalEntity).matchContent(
+                loadSampleData("expectations/subscription_with_notifications.jsonld")
+            )
+        )
+    }
+
+    @Test
+    fun `it should return a temporal entity without temporal attributes if it has no temporal history`() {
+        val temporalEntityAttribute = TemporalEntityAttribute(
+            entityId = "urn:ngsi-ld:Subscription:1234".toUri(),
+            type = "https://uri.etsi.org/ngsi-ld/Subscription",
+            attributeName = "https://uri.etsi.org/ngsi-ld/notification",
+            attributeValueType = TemporalEntityAttribute.AttributeValueType.ANY
+        )
+        val attributeAndResultsMap = mapOf(
+            temporalEntityAttribute to emptyList<AttributeInstanceResult>()
+        )
+
+        val temporalEntity = temporalEntityService.buildTemporalEntity(
+            "urn:ngsi-ld:Subscription:1234".toUri(),
+            attributeAndResultsMap,
+            listOf(NGSILD_CORE_CONTEXT),
+            false
+        )
+        assertTrue(
+            serializeObject(temporalEntity).matchContent(
+                loadSampleData("expectations/subscription_without_temporal_attributes.jsonld")
```

Comment:
shouldn't the attribute still be present but with an empty list of values?

Reply:
Not sure about this point of the spec, but i think that it makes more sense to be present with an empty list of values.
Updated [here](https://github.com/stellio-hub/stellio-context-broker/pull/327/commits/0e57a3226a98373f5de454bd35e99b527b9d58ee).